### PR TITLE
Remove v5 JSON session support

### PR DESCRIPTION
**Pull request by NickMeves**
Fully remove support for v5 JSON sessions that were replaced by lz4 compressed MessagePack in v6.

## Motivation and Context

Simplify the codebase and remove the complex logic needed for backwards compatibility that is no longer necessary.

## How Has This Been Tested?

Unit Tests

## Checklist:

- [x] My change requires a change to the documentation or CHANGELOG.
- [x] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 ## Important Notes
 
+- [#905](https://github.com/oauth2-proxy/oauth2-proxy/pull/905) Existing sessions from v6.0.0 or earlier are no longer valid. They will trigger a reauthentication.
```
Comment (JoelSpeed) [inline CHANGELOG.md:7]: Is this message accurate? I thought the sessions in v6.0.0 included the new format? Should it not say from before v6.0.0?

**Review by JoelSpeed — COMMENTED** [review]
Lgtm, just want some clarification on the changlog note

File: CHANGELOG.md (hunk 1 above)
Comment (NickMeves) [reply CHANGELOG.md:7]: I looked through the release notes history:

v6.0.0 got the SHA1 -> SHA256 change.
v6.1.0 got the MessagePack LZ4 sessions

All the code referred to it as `V5LegacySessions` but I think it missed the v6 major release and we finished it up by the minor release.

Comment (JoelSpeed) [reply CHANGELOG.md:7]: Ack, makes sense, shall we rebase and get this merged then

Comment (NickMeves) [reply CHANGELOG.md:7]: Just rebased 👍

Comment (NickMeves) [reply CHANGELOG.md:7]: I had so many unit tests for the legacy route.  Would be a shame if this lowers our testing coverage score 🤣

**Review by JoelSpeed — APPROVED** [review]
LGTM

**Pull request merged**